pandas/core/period.py

```diff
@@ -0,0 +1,469 @@
+
+    def __iter__(self):
+        return iter(self._data)
+
+    def __len__(self):
+        return len(self._data)
+
+    def __repr__(self):
+        values = self._format_values()
+        return "PeriodArray({}, freq={}, dtype={})".format(
+            values, self.freq, self.dtype
+        )
+
+    def __getitem__(self, item):
+        if is_scalar(item):
+            return self._box_func(self._data[item])
+        else:
+            values = self._data[item]
+            return self._simple_new(values, self.freq)
+
+    @property
+    def dtype(self):
+        if self._dtype is None:
+            self._dtype = PeriodDtype(self.freq)
+        return self._dtype
+
+    @property
+    def shape(self):
+        return (len(self),)
+
+    @property
+    def values(self):
+        return self.astype(object)
+
+    @property
+    def asi8(self):
+        return self._data.view('i8')
+
+    @property
+    def _box_func(self):
+        return lambda x: Period._from_ordinal(ordinal=x, freq=self.freq)
+
+    def _format_values(self):
+        return np.array(['%s' % x for x in self.values], dtype='object')
+
+    def formatting_values(self):
+        return self._format_values()
+
+    def astype(self, dtype, copy=True, how='start'):
+        dtype = pandas_dtype(dtype)
+        if is_object_dtype(dtype):
+            return np.array([Period._from_ordinal(p, self.freq)
+                             for p in self], dtype='object')
+        else:
+            raise ValueError('invalid dtype')
+
+    def copy(self):
+        return self._from_ordinals(self._data.copy(), freq=self.freq)
+
+    def isna(self):
+        return self.asi8 == iNaT
+
+    def nbytes(self):
+        return self._data.nbytes
+
+    def take(self, indexer, allow_fill=True, fill_value=None):
+        # XXX: is take supposed to be a view?
+        return self._from_ordinals(self._data.take(indexer), self.freq)
+
+    take_nd = take
+
+    @classmethod
+    def concat_same_type(cls, to_concat):
+        dtype = to_concat[0].dtype
+        if not all(other.dtype == dtype for other in to_concat):
+            raise TypeError("All frequencies must match")
+        values = np.concatenate([other._data for other in to_concat])
+        return cls._from_ordinals(values, freq=to_concat[0].freq)
+
+    def get_values(self):
+        return self._data
+
+    @classmethod
+    def _scalar_data_error(cls, data):
+        # TODO: array-mixin
+        raise TypeError('{0}(...) must be called with a collection of some '
+                        'kind, {1} was passed'.format(cls.__name__,
+                                                      repr(data)))
+
+    def _get_attributes_dict(self):
+        # TODO: from indexes.base, needed for ops, can remove
+        return {}
+
+    def view(self, cls=None):
+        return self._data.view(cls)
+
+    def equals(self, other):
+        if not isinstance(other, type(self)):
+            return False
+        return (self.freq == other.freq and
```

Comment:
`self.freq == other.freq` may raise, in which case you generally want to return False here.